vignettes/MSnbase-centroiding.Rmd

```diff
@@ -0,0 +1,333 @@
+m/z range from 105 to 130 and a retention time from 0 to 240 seconds. For more
+details on the sample see `?msdata::sciexdata`. Below we load the required
+libraries and read the MS data.
+
+```{r libraries, message = FALSE}
+library(MSnbase)
+library(msdata)
+
+fl <- dir(system.file("sciex", package = "msdata"), full.names = TRUE)[2]
+data_prof <- readMSData(fl, mode = "onDisk", centroided = FALSE)
+```
+
+We next extract the profile MS data for the [M+H]+ adduct of serine with the
+expected m/z of 106.049871. We thus filter the `data_prof` object using an m/z
+range containing the signal for the metabolite and a retention time window from
+175 to 187 seconds corresponding to the time when the analyte elutes from the
+LC. Subsequently we plot the profile MS data for serine.
+
+```{r serine-msdata, fig.align = "center", fig.cap = "MS profile data for serine. Upper panel shows the base peak chromatogram (BPC), lower panel the individual signals in the retention time - m/z space. The horizontal dashed red line indicates the theoretical m/z of the [M+H]+ adduct of serine."}
+## Define the mz and retention time ranges
+serine_mz <- 106.049871
+mzr <- c(serine_mz - 0.01, serine_mz + 0.01)
+rtr <- c(175, 187)
+
+## Filtering the object
+serine <- filterMz(filterRt(data_prof, rt = rtr), mz = mzr)
+
+plot(serine, type = "XIC")
+abline(h = serine_mz, col = "red", lty = 2)
+```
+
+The lower panel in the plot above shows the individual signal intensities
+measured by the mass spectrometer, the upper part the *base peak chromatogram*
+(BPC), which represents the maximum signal for each discrete retention time
+across the range of m/z values. The horizontal rows of points in the lower panel
+indicate the resolution of the mass spectrometer. Each column of data points
+(for a discrete time point) represents the signal for the ion in one
+spectrum. Below we plot the signal for one of the spectra containing signal for
+serine.
+
+```{r serine-spectrum24, fig.align = "center", fig.cap = "On of the spectra for serine in profile mode."}
+plot(serine[[22]])
+```
+
+The MS instrument recorded a signal along the m/z range in discrete intervals
+(which depend on the resolution of the instrument). The profile-mode signal of
+the serine ion at the respective retention time (the *mass peak*) consists
+therefore of multiple intensities that follow approximately a gaussian
+distribution.
+
+As detailed in the introduction, centroiding aims to reduce this signal
+distribution to a single representative intensity for the ion in a spectrum
+(i.e. at a given retention time). The simplest approach selects the largest
+intensity for each mass peak and report its intensity and m/z value. This can be
+done using the `pickPeaks` method with default parameters as shown below.
+
+```{r simple-pickPeaks, fig.align = "center", warning = FALSE, fig.cap = "Centroided data for serine."}
+
+data_cent <- pickPeaks(data_prof)
```

Comment:
Why don't you directly do 
```r
pickPeaks(serine)
```
here, rather then picking on the whole data and filtering again?

Reply:
first because I didn't want to make the impression that centroiding should be done on a per-ion level but rather on the full file. second because I'll use the `data_cent` also later for proline.